include/wil/resource.h

```diff
@@ -3454,6 +3463,12 @@ namespace wil
     template<typename T>
     using unique_virtualalloc_ptr = wistd::unique_ptr<T, virtualalloc_deleter>;
 
+    /** Manages a typed pointer allocated with MapViewOfFile
+    A specialization of wistd::unique_ptr<> that frees via UnmapViewOfFile(p).
+    */
+    template<typename T>
+    using unique_mapview_ptr = wistd::unique_ptr<T, mapview_deleter>;

```

Comment:
It'd be neat to also have something in win32_helpers.h which made `MapViewOfFile` easier, like:

```cpp
HRESULT map_file_view_nothrow(HANDLE file, DWORD access, const UINT64& offset, size_t size, unique_mapview& view)
{
    ULARGE_INTEGER offsetInt;
    offsetInt.QuadPart = offset;
    view.reset(::MapViewOfFile(file, access, offsetInt.LowPart, offsetInt.HighPart, size));
    RETURN_LAST_ERROR_IF_NULL(view);
    return S_OK;
}
unique_mapview map_file_view(HANDLE file, DWORD access, const UINT64& offset)
{
    unique_mapview view;
    THROW_IF_FAILED(map_file_view_nothrow(file, access, offset));
    return view;
}
```
And nothrow ... and maybe the `_ptr<T>` variant.